src/client/Microsoft.Identity.Client/Platforms/iOS/BrokerKeyHelper.cs

```diff
@@ -44,7 +44,18 @@ internal static byte[] GetRawBrokerKey(ICoreLogger logger)
                     ValueData = byteData
                 };
 
-                var result = SecKeyChain.Add(record);
+                var result = SecKeyChain.Add(secRecord);
+
+                if (result == SecStatusCode.DuplicateItem)
```

Comment:
if the above key is null, we try to add the record created earlier, but the two keys (Service and Account) are the same, so a DuplicateItem error is throw. If we try to do `SecStatusCode.Update(record, secRecord)`, there is a new error because secRecord has a few field -> ValueData, so it's best to just delete the original record and add the new one. I was able to repro this by setting the key to null.